### ISSUE TITLE
Update super-linter version and reinstate GitHub Action linting

### ISSUE THREAD

**Issue by stuartleeks** [microsoft/AzureTRE#1723]
## Description

_tldr - waiting on a super-linter release after v4.9.2 [here](https://github.com/github/super-linter/releases)_

While working on #1721, the latest version of `super-linter` had a version of `actionlint` that was incorrectly flagging errors relating to outputs from the GitHub script step. 

The [`actionlint` issue](https://github.com/rhysd/actionlint/issues/104) was resolved in 1.6.11 of `actionlint`.

At the time of creating this issue, the `super-linter` code on `main`  had been updated to 1.6.12 of `actionlint`, but the latest released `super-linter` image only seems to have `actionlint` 1.6.10.

(Tracked [here](https://github.com/github/super-linter/issues/2816) on `super-linter`)

Note: in the meantime, the `.github/scripts/run-test.sh` script includes the `actionlint` checks)

## Acceptance criteria

- Update  the `super-linter` version
- Set `VALIDATE_GITHUB_ACTIONS: true` in `build_validation_develop.yml`
- Linting passes for the workflows (can run `make lint` locally)

**Comment by tamirkamara** [comment]
@stuartleeks 4.9.3 is now out and I tested and saw that within superlinter all actions are passing. I thought to remove the linting step from the run-test.sh script, but it also doesn't heart keeping it there... thoughts?

**Comment by stuartleeks** [comment]
I'm in favour of keeping the linter in `run-test.sh` as I found that it gives a very tight feedback loop when developing locally 😄

**Comment by tamirkamara** [comment]
Thanks @stuartleeks . Closing per prior conversation above.

**Comment by stuartleeks** [comment]
I think it's still worth updating the version of super-linter used in CI and re-enabling the GH actions linting there? I see the actionlint in `run-tests.sh` as a developer convenience vs the safety net of checks in CI
